app/models/service_provider_request.rb

```diff
@@ -1,23 +1,20 @@
-class ServiceProviderRequest < ApplicationRecord
-  def self.create(attributes)
-    attributes[:loa] = attributes[:ial]
-    super
-  end
+class ServiceProviderRequest
+  attr_accessor :uuid, :issuer, :url, :loa, :requested_attributes
 
-  def self.from_uuid(uuid)
-    record = find_by(uuid: uuid) || NullServiceProviderRequest.new
-    record.ial = record.loa if record && !record.instance_of?(NullServiceProviderRequest)
-    record
-  rescue ArgumentError # a null byte in the uuid will raise this
-    NullServiceProviderRequest.new
+  def initialize(attrs = {})
+    @uuid = attrs[:uuid]
```

Comment:
What if these were all just keywords arguments? It would provide some nice typo protection. We could provide `nil` default values if needed? (Maybe default to empty array for `requested_attributes`):

```ruby
def initialize(uuid: nil, issuer: nil, url: nil, loa: nil, requested_attributes: [])
```